File: 品味Java的21種設計模式/dp_java/bridge/example7/MessageSMS.java
package cn.javass.dp.bridge.example7;
/**
 * 以站内短消息的方式发送消息
 */
public  class MessageSMS implements MessageImplementor{

	public void send(String message, String toUser) {
		System.out.println("使用站内短消息的方式，发送消息'"+message+"'给"+toUser);
	}
}

File: 品味Java的21種設計模式/dp_cpp/bridge/example6/MessageSMS.cpp
#include "MessageSMS.h"

namespace cn {
namespace javass {
namespace dp {
namespace bridge {
namespace example6 {

void MessageSMS::send(std::string message, std::string toUser) {
    puts("使用站内短消息的方式，发送消息'" + message + "'给" + toUser);
}
}
}
}
}
}
